### PR TITLE
Add special properties starting with "@" to additionaldata

### LINKED ISSUE #145: Add properties with '@' character to the AdditionalProperties bag
Looking at the C# snippet below from [here](https://docs.microsoft.com/en-us/graph/api/driveitem-post-children?view=graph-rest-1.0&tabs=csharp#request)

```cs

GraphServiceClient graphClient = new GraphServiceClient( authProvider );

var driveItem = new DriveItem
{
	Name = "New Folder",
	Folder = new Folder
	{
	},
	@microsoft.graph.conflictBehavior = "rename"
};

await graphClient.Me.Drive.Root.Children
	.Request()
	.AddAsync(driveItem);
```

The "@microsoft.graph.conflictBehavior" should be added to the AdditionalData bag as follows,

```cs
GraphServiceClient graphClient = new GraphServiceClient( authProvider );

var driveItem = new DriveItem
{
	Name = "New Folder",
	Folder = new Folder
	{
	},
	AdditionalData = new Dictionary<string, object>
	{
	    { "@microsoft.graph.conflictBehavior", "rename" }
	}
};

await graphClient.Me.Drive.Root.Children
	.Request()
	.AddAsync(driveItem);
```

This should be similar to the way we add "@odata" properties to the AdditionalData bag [here](https://github.com/microsoftgraph/microsoft-graph-explorer-api/blob/bdc2d5ebfb7a1ded56d4c356074cb220c14ecf63/CodeSnippetsReflection/LanguageGenerators/CSharpGenerator.cs#L347).

This is related to https://github.com/microsoftgraph/microsoft-graph-docs/pull/6085

### PR DESCRIPTION
**Pull request by andrueastman**
This PR closes #145 

#### Change proposed

Snippets with properties starting with the '@' character should also be added to the Additional data bag.

Therefore the snippet at this [link ](https://docs.microsoft.com/en-us/graph/api/driveitem-post-children?view=graph-rest-1.0&tabs=csharp#request)should look like this

```cs

GraphServiceClient graphClient = new GraphServiceClient( authProvider );

var driveItem = new DriveItem
{
	Name = "New Folder",
	Folder = new Folder
	{
	},
	AdditionalData = new Dictionary<string, object>
	{
	    { "@microsoft.graph.conflictBehavior", "rename" }
	}
};

await graphClient.Me.Drive.Root.Children
	.Request()
	.AddAsync(driveItem);

```

### CODE REVIEW

File: CodeSnippetsReflection.Test/CSharpGeneratorShould.cs [hunk 1]
```diff
@@ -805,5 +805,45 @@ public void GeneratesSnippetsWithOverridenTypeInBody()
             //Assert the snippet generated is as expected
             Assert.Equal(AuthProviderPrefix + expectedSnippet, result);
         }
+
+        [Fact]
+        // This tests asserts that a type beginning wirth "@" character is also added to the AdditionalData bag
```
Comment (irvinesunday) [inline CodeSnippetsReflection.Test/CSharpGeneratorShould.cs]: [nitpick] typo `wirth`

File: CodeSnippetsReflection/LanguageGenerators/CSharpGenerator.cs [hunk 2]
```diff
@@ -336,15 +336,15 @@ private static string CSharpGenerateObjectFromJson(ODataPathSegment pathSegment,
         }
 
         /// <summary>
-        /// Generates language specific code relate to various odata properties
+        /// Generates language specific code relate to add special properties to the AdditionalData dictionary
```
Comment (irvinesunday) [inline CodeSnippetsReflection/LanguageGenerators/CSharpGenerator.cs]: [nitpick] check grammar :-)

File: CodeSnippetsReflection.Test/CSharpGeneratorShould.cs (hunk 1 above)
Comment (andrueastman) [reply CodeSnippetsReflection.Test/CSharpGeneratorShould.cs]: Resolved

File: CodeSnippetsReflection/LanguageGenerators/CSharpGenerator.cs (hunk 2 above)
Comment (andrueastman) [reply CodeSnippetsReflection/LanguageGenerators/CSharpGenerator.cs]: Resolved

**Pull request merged**